pkg/kind/config.go

```diff
@@ -0,0 +1,90 @@
+package kind
+
+import (
+	"embed"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/cnoe-io/idpbuilder/api/v1alpha1"
+)
+
+type PortMapping struct {
+	HostPort      string
+	ContainerPort string
+}
+
+type TemplateConfig struct {
+	v1alpha1.BuildCustomizationSpec
+	KubernetesVersion string
+	ExtraPortsMapping []PortMapping
+	RegistryConfig    string
+}
+
+//go:embed resources/* testdata/custom-kind.yaml.tmpl
+var configFS embed.FS
+
+type HttpClient interface {
+	Get(url string) (resp *http.Response, err error)
+}
```

Comment:
I think we can move this to cluster.go and make it one of fields in the Cluster struct.

It would look something like:

```go
type Cluster struct {
	provider          IProvider
	httpClient        HttpClient
	name              string
	kubeVersion       string
	kubeConfigPath    string
	kindConfigPath    string
	extraPortsMapping string
	registryConfig    []string
	cfg               v1alpha1.BuildCustomizationSpec
}
```

In here: https://github.com/cnoe-io/idpbuilder/blob/64d93ace30b9700c81920c091ff24926066106c0/pkg/kind/cluster.go#L152-L170

We can call `util.GetHttpClient()`. When we test, use the mock client.